src/portable_python/__init__.py

```diff
@@ -632,8 +654,15 @@ def compile(self):
                         self._finalize()
                         return
 
-                # Split on '#' for urls that include a checksum, such as #sha256=... fragment
-                basename = runez.basename(self.url, extension_marker="#")
+                # Some URL's may not end in file extension, such as with redirects.
+                # Github releases asset endpoint is this way .../releases/assets/48151
+                if not self.url.endswith((".zip", ".tar.gz")):
+                    suffix = self.src_suffix or ".tar.gz"
+                    basename = f"{self.m_name}-{self.version}.{suffix}"
+                else:
+                    # Split on '#' for urls that include a checksum, such as #sha256=... fragment
+                    basename = runez.basename(self.url, extension_marker="#")
```

Comment:
I think we need to split first on `#`, and then look for known extension

Reply:
Makes sense, please feel free to edit if you are available

Reply:
Corrected and added test to observe that a `#sha256=...` doesn't fool this into thinking url has no known extension.